std/signature/ecdsa/ecdsa.go

```diff
@@ -19,17 +20,45 @@ type PublicKey[Base, Scalar emulated.FieldParams] sw_emulated.AffinePoint[Base]
 //
 // We assume that the message msg is already hashed to the scalar field.
 func (pk PublicKey[T, S]) Verify(api frontend.API, params sw_emulated.CurveParams, msg *emulated.Element[S], sig *Signature[S]) {
+	qxBits, rbits, err := pk.checkParams(api, params, msg, sig)
+	if err != nil {
+		panic("ecdsa check params error")
+	}
+	for i := range rbits {
+		api.AssertIsEqual(rbits[i], qxBits[i])
+	}
+}
+
+// IsVerified modified from the Verify method, asserts that the signature sig verifies for the message msg and public
+// key pk. The curve parameters params define the elliptic curve.
+// We assume that the message msg is already hashed to the scalar field.
+// If the signature is valid, it returns 1; otherwise, it returns  0
+func (pk PublicKey[T, S]) IsVerified(api frontend.API, params sw_emulated.CurveParams, msg *emulated.Element[S], sig *Signature[S]) frontend.Variable {
+	qxBits, rbits, err := pk.checkParams(api, params, msg, sig)
+	if err != nil {
+		panic("ecdsa check params error")
+	}
+	verified := frontend.Variable(1)
+	for i := range rbits {
+		res := api.IsZero(api.Sub(rbits[i], qxBits[i]))
+		verified = api.And(verified, res)
+	}
+	return verified
+
+}
+
+func (pk PublicKey[T, S]) checkParams(api frontend.API, params sw_emulated.CurveParams, msg *emulated.Element[S], sig *Signature[S]) ([]frontend.Variable, []frontend.Variable, error) {
 	cr, err := sw_emulated.New[T, S](api, params)
 	if err != nil {
-		panic(err)
+		return nil, nil, err
```

Comment:
```suggestion
		panic(err)
```

Let's stick to the original here